src/Infrastructure/IntegrationReport/Reporter.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Infrastructure\IntegrationReport;
+
+use App\Domain\User\Organization;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+final class Reporter
+{
+    private bool $_hasNewErrors;
```

Comment:
```suggestion
    private bool $hasNewErrors;
```

Je ne sais pas pourquoi tu as utilisé le `_`, si c'est par convention ou autre, mais le préfixe `has` suffit pour comprendre qu'il s'agit d'un booléen avec une portée privée comme toutes les autres propriétés.

Reply:
C'est parce qu'il y a aussi la méthode `hasNewErrors()` qui est un getter pour cette propriété... Donc il faut que j'évite le clash de nom

Reply:
Alors il faudrait changer la méthode d'update, par exemple `setNewError` ou `updateNewError` ?

Reply:
? Il n'y a pas de méthode d'update, c'est un getter qui s'appelle `hasNewErors()` qui renvoie juste l'état de la variable `$_hasNewErrors`

J'aurais bien voulu faire un `public readonly $hasNewErrors` mais ça ne permet pas de "lever le drapeau" à l'intérieur de la classe (vu que c'est readonly)

Après je peux renommer l'attribut, genre `$newErrorsDetected` ou autre

Reply:
Bon j'ai renommé l'attribut en `haveNewErrorsBeenReceived`